go/stats/export.go

```diff
@@ -263,11 +312,39 @@ func (v *Duration) String() string {
 
 // IntFunc converts a function that returns
 // an int64 as an expvar.
-type IntFunc func() int64
+type IntFunc struct {
+	f    func() int64
+	help string
+}
+
+// NewIntFunc creates a new IntFunc instance and publishes it if name is set
+// All IntFuncs are exported to Prometheus as gauges and not counters.
+func NewIntFunc(name string, help string, f func() int64) *IntFunc {
+	i := &IntFunc{
+		f:    f,
+		help: help,
+	}
+
+	if name != "" {
+		publish(name, i)
+		publishPullIntFunc(i, name)
+	}
+	return i
+}
 
 // String is the implementation of expvar.var
-func (f IntFunc) String() string {
-	return strconv.FormatInt(f(), 10)
+func (intFunc IntFunc) String() string {
```

Comment:
Go style recommends using single letter names for simple types like this. I don't necessarily agree with it, but it's better to follow the guidelines :).